[DNM] Add AH-next runtime - the post-migration Asset Hub runtime

seadanda opened the pull request:
The migration of staking, governance, balances and other related pallets from the Relay Chain to Asset Hub is currently in development (part of the Plaza project and RFC-32).

We can already have the future state of Asset Hub ready to test against for ourselves and other stakeholders who will be integrating with it, without needing to have a completed migration. Therefore by having two asset hubs on Westend in the lead up to the transition allows us to have the old runtime and future runtime live at the same time, and crucially for the future runtime to be live far in advance of the migration being ready.

We want to the migration on Westend to be a proper dry run for Kusama and Polkadot, but want to have a testing target much before then.

TODO:
- [x] Update from master to get blocknumber provider changes for relevant pallets from https://github.com/paritytech/polkadot-sdk/issues/6297
- [x] Transplant recent changes to AH here
- [x] Merge in https://github.com/paritytech/polkadot-sdk/pull/5501
- [x] Grab BlockNumberProvider changes from individual branches
- [x] Update staking pallets types from https://github.com/paritytech/polkadot-sdk/pull/6034
- [x] Update to multi block elections https://github.com/paritytech/polkadot-sdk/pull/7282
- [x] Merge https://github.com/paritytech/polkadot-sdk/pull/7424
- [x] Deploy (https://github.com/paritytech/devops/issues/3604)
- [ ] Integrate the staking interface pallet when it's ready (https://github.com/paritytech/polkadot-sdk/pull/7658)

seadanda committed:
Add asset-hub-next on westend

seadanda committed:
Add staking (WIP)

seadanda committed:
Final impl to satisfy pallet-staking

seadanda committed:
Add gov (wip)

seadanda committed:
Gov config (WIP)

seadanda committed:
Fix last few wee bits

seadanda committed:
Merge branch 'master' into donal-ah-next-westend

seadanda committed:
Sort toml files

seadanda committed:
Merge updates from AH

seadanda committed:
Update to XCMv5 and fix small bits from merge

seadanda committed:
Make xcm tests fail less hard

seadanda commented:
/cmd fmt

Commit:
Update from seadanda running command 'fmt'

seadanda committed:
Merge branch 'master' into donal-ah-next-westend

seadanda committed:
Update from AH

seadanda committed:
Update lockfile

kianenigma committed:
Add multi-block-staking (#7358)

pulls #7282 into ah-next. 

Please see only changes in ah-next runtime. 

- [x] Zombienet + omni-node setup to quickly run ah-next?

---------

Signed-off-by: xermicus <cyrill@parity.io>
Signed-off-by: Alexandru Vasile <alexandru.vasile@parity.io>
Signed-off-by: Andrei Sandu <andrei-mihail@parity.io>
Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Signed-off-by: Iulian Barbu <iulian.barbu@parity.io>
Signed-off-by: EleisonC <ckalule7@gmail.com>
Signed-off-by: Cyrill Leutwiler <bigcyrill@hotmail.com>
Co-authored-by: Gonçalo Pestana <g6pestana@gmail.com>
Co-authored-by: Ankan <10196091+Ank4n@users.noreply.github.com>
Co-authored-by: command-bot <>
Co-authored-by: Guillaume Thiolliere <gui.thiolliere@gmail.com>
Co-authored-by: Alexander Theißen <alex.theissen@me.com>
Co-authored-by: xermicus <cyrill@parity.io>
Co-authored-by: Yuri Volkov <0@mcornholio.ru>
Co-authored-by: Maksym H <1177472+mordamax@users.noreply.github.com>
Co-authored-by: Santi Balaguer <santiago.balaguer@gmail.com>
Co-authored-by: Dónal Murray <donal.murray@parity.io>
Co-authored-by: PG Herveou <pgherveou@gmail.com>
Co-authored-by: Sebastian Kunert <skunert49@gmail.com>
Co-authored-by: Michal Kucharczyk <1728078+michalkucharczyk@users.noreply.github.com>
Co-authored-by: seemantaggarwal <32275622+seemantaggarwal@users.noreply.github.com>
Co-authored-by: Iulian Barbu <14218860+iulianbarbu@users.noreply.github.com>
Co-authored-by: Branislav Kontur <bkontur@gmail.com>
Co-authored-by: Ron <yrong1997@gmail.com>
Co-authored-by: runcomet <runcomet@protonmail.com>
Co-authored-by: Giuseppe Re <giuseppe.re@parity.io>
Co-authored-by: Benjamin Gallois <benjamin@gallois.cc>
Co-authored-by: José Molina Colmenero <jose@blockdeep.io>
Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: Clara van Staden <claravanstaden64@gmail.com>
Co-authored-by: Ludovic_Domingues <ludovic.domingues96@gmail.com>
Co-authored-by: Javier Viola <363911+pepoviola@users.noreply.github.com>
Co-authored-by: Serban Iorga <serban@parity.io>
Co-authored-by: Stephane Gurgenidze <59443568+sw10pa@users.noreply.github.com>
Co-authored-by: Mrisho Lukamba <69342343+MrishoLukamba@users.noreply.github.com>
Co-authored-by: Alexandru Vasile <60601340+lexnv@users.noreply.github.com>
Co-authored-by: FereMouSiopi <FereMouSiopi@proton.me>
Co-authored-by: Egor_P <egor@parity.io>
Co-authored-by: Alisher A. Khassanov <a.khssnv@gmail.com>
Co-authored-by: Alin Dima <alin@parity.io>
Co-authored-by: Andrei Sandu <54316454+sandreim@users.noreply.github.com>
Co-authored-by: cmd[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: Andrei Eres <eresav@me.com>
Co-authored-by: Dmitry Markin <dmitry@markin.tech>
Co-authored-by: dharjeezy <dharjeezy@gmail.com>
Co-authored-by: christopher k <30932534+EleisonC@users.noreply.github.com>
Co-authored-by: Alexander Samusev <41779041+alvicsam@users.noreply.github.com>
Co-authored-by: Francisco Aguirre <franciscoaguirreperez@gmail.com>
Co-authored-by: Adrian Catangiu <adrian@parity.io>
Co-authored-by: Javier Viola <javier@parity.io>
Co-authored-by: Andrew Jones <ascjones@gmail.com>
Co-authored-by: James Wilson <james@jsdw.me>
Co-authored-by: Guillaume Thiolliere <guillaume.thiolliere@parity.io>

seadanda commented:
/cmd fmt

github-actions[bot] committed:
Update from seadanda running command 'fmt'

seadanda committed:
Add chopsticks instructions

Ank4n committed:
[AHM] merge paged slashing into AH next (#7468)

Merges on-going [paged slashing
changes](https://github.com/paritytech/polkadot-sdk/pull/7424) into AH
next.

---------

Co-authored-by: Dónal Murray <donal.murray@parity.io>

seadanda commented:
/cmd fmt

github-actions[bot] committed:
Update from seadanda running command 'fmt'

seadanda committed:
Clean up a bit

niklasad1 commented on cumulus/parachains/runtimes/assets/asset-hub-next-westend/zombienet-omni-node.toml:
```diff
@@ -0,0 +1,22 @@
+
+[parachains.collator]
+name = "charlie"
+ws_port = 9988
```
```suggestion
rpc_port = 9988
```

niklasad1 commented on cumulus/parachains/runtimes/assets/asset-hub-next-westend/zombienet-omni-node.toml:
```diff
@@ -0,0 +1,22 @@
+[[relaychain.nodes]]
+name = "bob"
+validator = true
+ws_port = 9955
```
```suggestion
rpc_port = 9955
```

niklasad1 commented on cumulus/parachains/runtimes/assets/asset-hub-next-westend/zombienet-omni-node.toml:
```diff
@@ -0,0 +1,22 @@
+[[relaychain.nodes]]
+name = "alice"
+validator = true
+ws_port = 9944
```
```suggestion
rpc_port = 9944
```

seadanda committed:
Update zombienet-omni-node.toml

Co-authored-by: Niklas Adolfsson <niklasadolfsson1@gmail.com>

seadanda committed:
Update zombienet-omni-node.toml

Co-authored-by: Niklas Adolfsson <niklasadolfsson1@gmail.com>

seadanda committed:
Update zombienet-omni-node.toml

Co-authored-by: Niklas Adolfsson <niklasadolfsson1@gmail.com>

seadanda committed:
Merge branch 'master' into donal-ah-next-westend

seadanda committed:
Merge commit '5c9fda56c377e2818ab082412392ce06eea3f34f' into donal-ah-next-westend

seadanda committed:
Revert "[AHM] merge paged slashing into AH next (#7468)"

This reverts commit 019300361587665960356e7839660d33afef9e8f.

seadanda committed:
Fix last merge breakages

seadanda committed:
Clobber auto-merged stuff that was incorrect

seadanda committed:
Update from AH

seadanda committed:
Update AH-next to work with stable2503 and upstream AH

seadanda committed:
Fix tx payment

seadanda committed:
Remove unnecessary type hint

kianenigma committed:
EPMB patches from kian

kianenigma committed:
update instructions a bit more

kianenigma committed:
update instructions

seadanda committed:
Add more issuance to fix transaction payment issue

rockbmb committed:
Add note to `README` for AHNW (#7709)

Co-authored-by: Dónal Murray <donal.murray@parity.io>

kianenigma committed:
new one liner

kianenigma committed:
new batch of changes from kian -- ah-next still builds fine, just some warnings

kianenigma committed:
more changes, snapshot works in ah-next

kianenigma committed:
update instructions

seadanda committed:
Add sudo and root-offences and fix collator key

seadanda committed:
Temporarily remove root-offences due to Exposure mismatch

seadanda committed:
Get root-offences working

niklasad1 committed:
use rpc_port in zombienet-omni-node.toml

kianenigma committed:
ah-next compiles -- all good

kianenigma committed:
some small fixes, nothing big yet

bkontur committed:
Fix `Collectives` location for AHW-next configuration (#7975)

`<Junctions>.into_location()` generates `Location { parents: 0,
interior: <Junctions> }` - `parents` defaults to `0`

rossbulat commented:
This draft PR looks stale now. It is marked `[DNM]`, targets the post-migration Asset Hub runtime work, and has had no follow-up activity since March 2025.

I think we can close this PR. If any of this AH-next runtime work is still relevant, it would be better to revive it as a fresh PR against current `master`.
